File: py/validPar.py
def isValid(s: str) -> bool:
	map = {
		")": "(",
		"}": "{",
		"]": "["
	}
	stack = []
	
	for char in s:
		if char in map:
			if stack and map[char] == stack[-1]:
				stack.pop()
			else:
				return False
		else:
			stack.append(char)
	return not stack
	
test = "{}{}"
test2 = "()"
test3 = "(({}))"
test4 = "(]"

print(isValid(test), "true")
print(isValid(test), "true")
print(isValid(test), "true")
print(isValid(test), "false")

File: js/validPar.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let map = {
        ")": "(",
        "}": "{",
        "]": "["
    };

    let stack = [];
    for (let c of s){
        if (map.hasOwnProperty(c)){
            if (stack.length > 0 && stack[stack.length - 1] == map[c]){
                stack.pop();
                continue;
            }
            return false;
        }
        stack.push(c);
    }

    return stack.length == 0;
};
